amplify/function/BusinessLogic/ResetHackathon/handler.ts

```diff
@@ -0,0 +1,267 @@
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+
+import type { Schema } from "../../../data/resource";
+import type { ScoreComponentTypeInput } from "./graphql/API";
+import {
+  deleteScore,
+  deleteTeam,
+  deleteTeamRoom,
+  deleteUser,
+  updateHackathon,
+} from "./graphql/mutations";
+import {
+  listHackathons,
+  listScores,
+  listTeamRooms,
+  listTeams,
+  listUsers,
+} from "./graphql/queries";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: process.env.AMPLIFY_DATA_GRAPHQL_ENDPOINT as string,
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+type Handler = Schema["ResetHackathon"]["functionHandler"];
+
+export const handler: Handler = async (event) => {
+  try {
+    // Extract arguments from the event if any
+    const {
+      safetyCheck,
+      resetRooms,
+      resetScores,
+      resetTeams,
+      resetUsers,
+      startDate,
+      endDate,
+      scoreComponents,
+      scoringSidepots,
+    } = event.arguments;
+    if (safetyCheck !== process.env.EDIT_HACKATHON_CODE) {
+      return {
+        statusCode: 403,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+    // get the current hackathon model data items
+    const HackathonItems = (
+      await client.graphql({
```

Comment:
For each call to client, we must add a check to see if errors is present and then Throw if it is. This will ensure Errors are actually thrown to our try/catch block. This is similar to what we4 discussed in our last meeting for the frontend.